platform/android/src/android_renderer_frontend.cpp

```diff
@@ -70,13 +71,13 @@ AndroidRendererFrontend::~AndroidRendererFrontend() = default;
 
 void AndroidRendererFrontend::reset() {
     mapRenderer.reset();
-    rendererObserver.reset();
 }
 
 void AndroidRendererFrontend::setObserver(RendererObserver& observer) {
     assert (util::RunLoop::Get());
-    rendererObserver = std::make_unique<ForwardingRendererObserver>(*mapRunLoop, observer);
-    mapRenderer.actor().invoke(&Renderer::setObserver, rendererObserver.get());
+    // Don't call the Renderer directly, but use MapRenderer#setObserver to make sure
+    // the Renderer may be re-initialised without losing the RendererObserver reference.
+    mapRenderer.setObserver(std::make_unique<ForwardingRendererObserver>(*mapRunLoop, observer));
```

Comment:
@jfirebaugh This is a bit awkward, but I need the MapRenderer to be able to hold on to the Observer as the initialisation can happen in two orders (map observer first / gl surface initialisation first) and we need to deal with that using a mutex. Also, the MapRenderer needs the reference for re-initialisation after the app returns from the background so it can set it on the newly initialised renderer.

I kept the definition of the `ForwardingRendererObserver` as the `MapRenderer` has no access to the `RendererObserver` header.